server/health.go

```diff
@@ -56,15 +56,24 @@ func (s *healthService) Check(ctx context.Context, in *healthpb.HealthCheckReque
 		return &healthpb.HealthCheckResponse{Status: healthpb.HealthCheckResponse_UNKNOWN}, status.Errorf(codes.Internal, "failed to get clone status: err=%v", err)
 	}
 
-	var isOutOfSynced, isUnderCloning bool
-	if replicaStatus != nil && replicaStatus.LastIoErrno != 0 {
+	var isOutOfSynced, hasSQLThreadError, isUnderCloning bool
+	if replicaStatus != nil &&
+		replicaStatus.SlaveIOState != "" &&
+		(replicaStatus.LastIoErrno != 0 || replicaStatus.SlaveIORunning != moco.ReplicaRunConnect) {
 		isOutOfSynced = true
```

Comment:
How about this?
An IO thread error is one of the factors of OutOfSync, isn't it?

```suggestion
		hasIOThreadError = true
```